capstone/src/main/java/com/google/sps/data/Lesson.java

```diff
@@ -104,28 +104,40 @@ public Date getDate() {
-            case Lesson.TYPE_IMAGE:
-                return new Image(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
-                    jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Image.URL_PROPERTY_KEY).getAsString());
-            case Lesson.TYPE_CONTENT:
-                JsonElement listJson = jobject.get(Content.URLS_PROPERTY_KEY);
-                Type listType = new TypeToken<List<String>>() {}.getType();
-                List<String> urlList = new Gson().fromJson(listJson, listType);
-
-                return new Content(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
-                    jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Content.CONTENT_PROPERTY_KEY).getAsString(),
-                    urlList);
-            default:
-                throw new IllegalArgumentException(); 
+        String type = null;
+        try {
+            type = jobject.get(Lesson.TYPE_PROPERTY_KEY).getAsString();
+            if(type == null){
+                throw new IllegalArgumentException();
+            }
+        } catch (NullPointerException e){
+            throw new IllegalArgumentException(); 
+        }
+        try{
+            switch(type) {
+                case Lesson.TYPE_FORM:
+                    return new Form(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
+                        jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Form.EDIT_URL_PROPERTY_KEY).getAsString(),
+                        jobject.get(Form.URL_PROPERTY_KEY).getAsString());
+                case Lesson.TYPE_VIDEO:
+                    return new Video(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
+                        jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Video.URL_PROPERTY_KEY).getAsString());
+                case Lesson.TYPE_IMAGE:
+                    return new Image(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
+                        jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Image.URL_PROPERTY_KEY).getAsString());
+                case Lesson.TYPE_CONTENT:
+                    JsonElement listJson = jobject.get(Content.URLS_PROPERTY_KEY);
+                    Type listType = new TypeToken<List<String>>() {}.getType();
+                    List<String> urlList = new Gson().fromJson(listJson, listType);
+
+                    return new Content(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
+                        jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Content.CONTENT_PROPERTY_KEY).getAsString(),
+                        urlList);
+                default:
+                    throw new IllegalArgumentException(); 
+            }
+        } catch (NullPointerException e) {
+            throw new IllegalArgumentException();
```

Comment:
(optional/can be done in a later PR) these dont need to be in separate try blocks. They are both catching the same exception and are immediately consecutive, so they should be in a single block to make it easier to read

Reply:
Got it. My thought process was that they are catching exceptions at different times so they would each need their own try and catch block. This makes more sense though. I plan to keep note and fix in a later PR.